tsdb/shard.go

```diff
@@ -997,6 +1048,17 @@ func (s *Shard) ExpandSources(sources influxql.Sources) (influxql.Sources, error
 	return expanded, nil
 }
 
+// Backup backs up the shard by creating a tar archive of all TSM files that
+// have been modified since the provided time. See Engine.Backup for more details.
+func (s *Shard) Backup(w io.Writer, basePath string, since time.Time) error {
+	s.mu.RLock()
```

Comment:
This is going to block cache snapshotting now, when it does not need to.  Can you hold onto a reference for the engine so that that you don't need to hold a read lock while the backup is running?